app/src/main/java/org/systers/mentorship/view/fragments/TasksFragment.kt

```diff
@@ -43,49 +42,63 @@ class TasksFragment(private var mentorshipRelation: Relationship) : BaseFragment
-                if (successful) {
-                    if (taskViewModel.tasksList.isEmpty()) {
+            if (successful != null && successful) {
+                    if (taskViewModel.todoTasksList.isEmpty()) {
                         tvNoTask.visibility = View.VISIBLE
                         rvTasks.visibility = View.GONE
                     } else {
                         rvTasks.apply {
                             layoutManager = LinearLayoutManager(context)
-                            adapter = TasksAdapter(taskViewModel.tasksList, markTask)
+                            adapter = TasksAdapter(taskViewModel.todoTasksList, markTask)
                         }
                         tvNoTask.visibility = View.GONE
                     }
-                } else {
-                    view?.let {
-                        Snackbar.make(it, taskViewModel.message, Snackbar.LENGTH_LONG).show()
+
+                    if (taskViewModel.achievedTasksList.isEmpty()) {
+                        tvNoAchievements.visibility = View.VISIBLE
+                        rvAchievements.visibility = View.GONE
+                    } else {
+                        rvAchievements.apply {
+                            layoutManager = LinearLayoutManager(context)
+                            adapter = TasksAdapter(taskViewModel.achievedTasksList, markTask)
+                        }
+                        tvNoAchievements.visibility = View.GONE
                     }
+                } else {
+                view?.let {
+                    Snackbar.make(it, taskViewModel.message, Snackbar.LENGTH_LONG).show()
                 }
             }
         })
 
         fabAddItem.setOnClickListener {
-            showDialog()
+            showDialog(mentorshipRelation)
         }
 
         taskViewModel.getTasks(mentorshipRelation.id)
 
     }
 
     /**
-     * The function creates a dialog box through whoch new tasks can be added
+     * The function creates a dialog box through which new tasks can be added
      */
-    fun showDialog() {
+    fun showDialog(mentorshipRelation: Relationship) {
         val builder = AlertDialog.Builder(context)
         val inflater = layoutInflater
         builder.setTitle(appContext.getString(R.string.add_new_task))
         val dialogLayout = inflater.inflate(R.layout.dialog_add_task, null)
         val editText = dialogLayout.findViewById<EditText>(R.id.etAddTask)
         builder.setView(dialogLayout)
         builder.setPositiveButton(appContext.getString(R.string.save)) { dialogInterface, i ->
-            val newTask: String = editText.text.toString()
-            taskViewModel.addTask(newTask)
+            taskViewModel.addTask(mentorshipRelation.id, editText.text.toString())
+            taskViewModel.successfulCreate.observe(this, Observer {
+                successful ->
+                if (successful != null && successful) {
+                    taskViewModel.getTasks(mentorshipRelation.id)
```

Comment:
I just reload the list here, I thought it was easier than having to add the newly created task to the existing list. Is that okay?

Reply:
At first sight, I think it is ok, and if it can be improved, it can be later @jenpaff